tests/test_packages/test_skills/test_generic_buyer/test_behaviours.py

```diff
@@ -0,0 +1,312 @@
+            ledger_id=FETCHAI,
+            address=self.skill.skill_context.agent_address,
+        )
+        assert has_attributes, error_str
+
+    def test_search_behaviour_setup_not_is_ledger_tx(self):
+        """Test the setup method of the search behaviour where is_ledger_tx is False."""
+        # setup
+        strategy = cast(GenericStrategy, self.skill.skill_context.strategy)
+        strategy._is_ledger_tx = False
+        search_behaviour = cast(
+            GenericSearchBehaviour, self.skill.skill_context.behaviours.search
+        )
+
+        # before
+        assert not strategy.is_searching
+
+        # operation
+        search_behaviour.setup()
+
+        # after
+        assert strategy.is_searching
+
+    def test_search_behaviour_act_is_searching(self):
+        """Test the act method of the search behaviour where is_searching is True."""
+        # setup
+        search_behaviour = cast(
+            GenericSearchBehaviour, self.skill.skill_context.behaviours.search
+        )
+        strategy = cast(GenericStrategy, self.skill.skill_context.strategy)
+        strategy._is_searching = True
+
+        # operation
+        search_behaviour.act()
+
+        # after
+        assert self.get_quantity_in_outbox() == 1, "No message in outbox."
+
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=OefSearchMessage,
+            performative=OefSearchMessage.Performative.SEARCH_SERVICES,
+            to=self.skill.skill_context.search_service_address,
+            sender=self.skill.skill_context.agent_address,
+            query=self.skill.skill_context.strategy.get_location_and_service_query(),
+        )
+        assert has_attributes, error_str
+
+    def test_search_behaviour_act_not_is_searching(self):
+        """Test the act method of the search behaviour where is_searching is False."""
+        # setup
+        strategy = cast(GenericStrategy, self.skill.skill_context.strategy)
+        strategy._is_searching = False
+        search_behaviour = cast(
+            GenericSearchBehaviour, self.skill.skill_context.behaviours.search
+        )
+
+        # operation
+        search_behaviour.act()
+
+        # after
+        assert self.get_quantity_in_outbox() == 0
+
+
+class TestSkillHandler(BaseSkillTestCase):
+    """Test handlers of generic buyer."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_buyer")
+
+    def test_fipa_handler_handle_unidentified_dialogue(self):
+        """Test the _handle_unidentified_dialogue method of the fipa handler."""
+        # setup
+        fipa_handler = cast(GenericFipaHandler, self.skill.skill_context.handlers.fipa)
+        incoming_message = self.build_incoming_message(
+            message_type=FipaMessage,
+            dialogue_reference=("", ""),
+            performative=FipaMessage.Performative.ACCEPT,
+        )
```

Comment:
```suggestion
        invalid_dialogue_ref = ("", ""),
        incoming_message = self.build_incoming_message(
            message_type=FipaMessage,
            dialogue_reference=invalid_dialogue_ref,
            performative=FipaMessage.Performative.ACCEPT,
        )
```
makes it easier for the next developer to understand

Reply:
Done